packages/jest-matchers/src/index.js

```diff
@@ -78,9 +84,27 @@ const addMatchers = (matchersObj: MatchersObject): void => {
   Object.assign(global[GLOBAL_MATCHERS_OBJECT_SYMBOL], matchersObj);
 };
 
+const _validateResult = result => {
```

Comment:
true. i'd be interested in benchmarking this stuff. I don't think it's going to affect the overall performance that much though.